### DISCUSSION TITLE
Re-use NOMAD OASIS Parsers for Harvesters

### DISCUSSION THREAD

**Discussion by apirogov** [Materials-Data-Science-and-Informatics/metador-core#65] [Ideas]
Existing parsers for NOMAD maybe could be re-used for extracting metadata from unstructured/proprietary formats into the NOMAD schema system.

A Metador harvester then basically just would need to repackage the metadata into a Metador schema.

Ideally this would be supported by a way to derive Metador schema (drafts) from NOMAD MetaInfo automatically to minimize boilerplate work.

https://nomad-lab.eu/prod/rae/docs/parser.html

**Comment by broeder-j** [comment]
For the NOMAD oasis there was a change in how they do it:

1. Before: they wrote there own parsers, all depend on NOMAD.
2. Now: Parsers have to be their own standalone package. Details, how they now ensure the common Metadata I do not know.

The parsers from 2 can probably easily be re-used. the first one probably not.

**Reply by apirogov** [reply]
Yes, I guess I mean 2. @strupp-a made me aware of that possibility.

See that's why Metador is a collection of plugins to begin with - simplify reuse ;)
